eoflow/models/segmentation.py

```diff
@@ -240,6 +241,7 @@ def train_and_evaluate(self,
                            num_epochs,
                            iterations_per_epoch,
                            model_directory,
+                           validation_steps=1,
```

Comment:
which defaults should we use? 

by using `None` I was getting an error that step is required for infinite tf datasets

Reply:
Oh, my bad. Didn't realize at first you added the `validation_steps` parameter. I was thinking about `iterations_per_epoch`.

In your case the problem seems to be, that the `val_dataset` you provided is infinite in which case the evaluation can't be performed. For the validation always a finite dataset must be given. It's better to resolve this on the dataset end not here (by ceating a finite dataset or calling `dataset.take(number_of_examples)`). You can call `train_and_evaluate` with the `validation_steps` parameter if other options are not available, but it should not have a default value.

Reply:
so you suggest removing the `validation_steps` altogether?

Reply:
Yeah I think that would be the best approach. It can still be specified manualy if needed, as it will be passed through `**kwargs` to the `fit` method.

Reply:
OK, `validation_steps` removed, and made `validation_set` not repeating